fix(components): [message-box] Allow set action for ElMessageBox.close

lzjyzq2 opened the pull request:
Allow set action for `ElMessageBox.close`
- Added support for setting close actions to allow callback and promise execution after closing
- Fixed an issue where executing close would not execute a promise

closed #12363

Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

lzjyzq2 committed:
fix(components): [message-box] Modify default value of `state.action'

Add default value to `state.action` to prevent callback from not executing when MessageBox is closed

closed #12363

abichinger commented:
I think the best solution would be to change the signature of `ElMessageBox.close` to

```ts
ElMessageBox.close(action?: Action): void
```

lzjyzq2 committed:
feat(components): [message-box] allow set action for ElMessageBox.close

- Added support for setting close actions to allow callback and promise execution after closing
- Fixed an issue where executing close would not execute a promise

closed #12363
